build/build.go

```diff
@@ -548,12 +558,28 @@ func (s *Session) BuildFiles(filenames []string, pkgObj string, cwd string) erro
 		return fmt.Errorf("named files must all be in one directory; have: %v", strings.Join(dirList, ", "))
 	}
 
+	root := dirList[0]
+	ctx := build.Default
+	ctx.UseAllFiles = true
+	ctx.ReadDir = func(dir string) (infos []fs.FileInfo, err error) {
+		for _, f := range filenames {
```

Comment:
nit: `infos` could be preallocated here, if we remove the named return values.

Reply:
fixed